go/vt/wrangler/keyspace.go

```diff
@@ -553,6 +552,13 @@ func (wr *Wrangler) masterMigrateServedType(ctx context.Context, keyspace string
 	// Always setup reverse replication. We'll start it later if reverseReplication was specified.
 	// This will allow someone to reverse the replication later if they change their mind.
 	if err := wr.setupReverseReplication(ctx, sourceShards, destinationShards); err != nil {
+		// It's safe to unfreeze if reverse replication setup fails.
+		wr.cancelMasterMigrateServedTypes(ctx, sourceShards)
```

Comment:
There's one more corner case to handle here: if setupReverseReplication partially succeeded, it's possible that it would have setup source shards for the reverse. If so, there will be no easy way to tell which way the replication is going, because both sides will have source shards.

If you look in `findSourceDest`, the presence a frozen flag is used as the tie breaker to disambiguate.

This means that we have to 'remove the source shards from the source shards' (so confusing) before unfreezing. I think we can do this inside the `cancelMasterMigrateServedTypes` function. It currently calls `updateShardRecords`. I recommend expanding that out within the function and additionally setting `si.SourceShards` to nil.

Reply:
Done, I think